### PR TITLE
remaining change from #4205

### PR DESCRIPTION
**Pull request by i-walker**
There was no need for further functions only those modifiers

### CODE REVIEW

**Comment by i-walker** [comment]
I excuse my question, but I can't debug the CI failures, due to no access :) 
@NikolajBjorner Hope those changes are fine. 
Thanks a lot a priori.

**Comment by NikolajBjorner** [comment]
The Z3Build-CI are not visible, unfortunately, because of some dependencies on signing binaries. 
The Travis builds, on the other hand are visible.
If you follow the links to the travis build you will see:

/home/user/z3_build/CMakeFiles/z3JavaJar.dir/java_compiled_z3JavaJar
/home/user/z3_src/src/api/java/Lambda.java:104: error: illegal start of expression
	public long nativeObject = Native.mkLambda(ctx.nCtx(),
	^
1 error
ninja: build stopped: subcommand failed.
The command '/bin/sh -c ${Z3_SRC_DIR}/contrib/ci/scripts/build_z3_cmake.sh' returned a non-zero code: 1
/home/travis/.travis/functions: line 607:  4480 Terminated    

In other words, your code change doesn't even compile.
It helps to compile locally (in your case with the build options for Java turned on) before round-tripping to the CI.

Independently of this, I am not sure about this pull request. It exposes methods that are private
and should by the intent of @wintersteiger not be exposed. Is there a reason for exposing these methods?
